src/visual.ts

```diff
@@ -1557,13 +1575,15 @@ module powerbi.extensibility.visual {
                         mekko.interactivityService,
                         true);
 
-                    mekko.categoryLegends[index] = legend;
+                    mekko.categoryLegends[index] = <ILegend>legend;
                 }
             });
 
-            legendParentsWithData.exit().remove();
-            let svgHeight: number = 26;
             if (reducedLegends.length > 0) {
+                this.categoryLegends.forEach((legend, index) => {
+                    (<ILegendGroup>legend).position = +d3.select((<ILegendGroup>legend).element).style("top").replace("px", "");
```

Comment:
Legend actually implements method called getMargin that returns current offset. Can we use this method instead?